httpcore5-h2/src/main/java/org/apache/hc/core5/http2/config/H2Config.java

```diff
@@ -121,8 +121,7 @@ public String toString() {
                 .setPushEnabled(INIT_ENABLE_PUSH)
                 .setMaxConcurrentStreams(Integer.MAX_VALUE) // no limit
                 .setMaxFrameSize(INIT_MAX_FRAME_SIZE)
-                .setInitialWindowSize(INIT_WINDOW_SIZE)
-                .setHeaderTableSize(Integer.MAX_VALUE); // unlimited
+                .setInitialWindowSize(INIT_WINDOW_SIZE);
```

Comment:
@rschmitt `.setHeaderTableSize(Integer.MAX_VALUE); // unlimited` should be `.setMaxHeaderListSize(Integer.MAX_VALUE); // unlimited`

Mea culpa

Reply:
Fair enough, but this option is not actually implemented.

Reply:
> Fair enough, but this option is not actually implemented.

I am presently working on it unless you want me to focus on something more important blocking the GA release.

Reply:
I don't have a root cause or even a detailed problem description yet, but I've observed that `SubscribeToShardIntegrationTest` fails when `H2Config#setInitialWindowSize` is called with a non-default window size. In fact, even setting the initial window size to 65,534 (i.e. one less than the default) seems to consistently result in `FLOW_CONTROL_ERROR`s from the server. This is the most severe bug I'm currently able to reproduce. (The `IOSession` shutdowns are even more concerning, but I don't have a repro case at the moment.)